src/crystal/system/unix/evented/waiters.cr

```diff
@@ -62,4 +66,16 @@ struct Crystal::Evented::Waiters
       {% end %}
     end
   end
+
+  # Dequeues all pending events and marks the list as closed. This must be
+  # called when a fd is closed or an error or hup event occurred.
+  def ready_all(& : Pointer(Event) ->) : Nil
```

Comment:
**nitpick:** Maybe this method name could be more explicit that readying all means the fd is closed. What about `#close_all`?

Reply:
We don't know if the fd is closed. It may have been explicitly closed or in an error state (yet not closed) or other end closed but there might still be things to read on our hand (hup). Except for the explicit close, other cases may or may not be closed.

One thing though: the `fd` won't block anymore (until it's reopened and reused).

Reply:
I was actually referring to the `@closed` state of `Waiters`. Maybe `#ready_all_and_close` would be more descriptive?
Anyway, it's just a nitpick.

Reply:
Maybe `@closed` could be better named, for example `@always_ready` :thinking: